### PR TITLE
Examples: Updates for color management (pt2)

### PR DESCRIPTION
**Pull request by donmccurdy**
Related:

- #23614

### CODE REVIEW

File: examples/jsm/loaders/3MFLoader.js [hunk 1]
```diff
@@ -359,8 +359,7 @@ class ThreeMFLoader extends Loader {
 				const colorNode = colorNodes[ i ];
 				const color = colorNode.getAttribute( 'color' );
 
-				colorObject.setStyle( color.substring( 0, 7 ) );
-				colorObject.convertSRGBToLinear(); // color is in sRGB
+				colorObject.setStyle( color.substring( 0, 7 ), SRGBColorSpace );
```
Comment (donmccurdy) [inline examples/jsm/loaders/3MFLoader.js:362]: Note: SRGBColorSpace is the default — this is just to be explicit that we know the loader's colors are sRGB. Would also be fine with removing it. See next comment.

File: examples/jsm/loaders/LDrawLoader.js
Comment (donmccurdy) [inline examples/jsm/loaders/LDrawLoader.js:1]: @gkjohnson I think you'd expressed an opinion about this in the past, so just to run it by you — with ColorManagement enabled, three.js understands the three.js strings used here are sRGB, and there's no need to convert explicitly. Do you feel OK about removing the convertSRGBToLinear() lines below?

The alternative is ... a bit confusing:

```
material = new MeshStandardMaterial();
material.color.setStyle( color, LinearSRGBColorSpace );
material.color.convertSRGBToLinear();
```

We specify LinearSRGBColorspace (or NoColorSpace?) to ensure ColorManagement does *not* do anything with the given (sRGB) value. Then we explicitly convert it from sRGB to Linear.

I think I prefer what I'm doing here, but it does mean the loader returns different data with ColorManagement disabled. That could be a bad thing, or the whole point of the flag, depending on your perspective...

Comment (gkjohnson) [reply examples/jsm/loaders/LDrawLoader.js:1]: I think I've come around on this. My concern is primarily around the clarity of these entries and the explicit use of the correct color space when reading it from a file format. To that end this is most clear to me but I know it's more verbose:

```js
material = new MeshStandardMaterial( { color: new Color( hex, SRGBColorSpace ), ... } );
```

Or maybe even just a comment somewhere stating that colors in the original file format are expected to be in SRGB so it's clear what's going on when reviewing the code. I don't want to block this, though.

File: examples/jsm/loaders/3MFLoader.js (hunk 1 above)
Comment (donmccurdy) [reply examples/jsm/loaders/3MFLoader.js:362]: I'm concerned that down the road (a) our understanding of a format's specification may change, or (b) our approach to input color space may change. For example, our current defaults...

```js
color.setRGB( r, g, b, ColorManagement.workingColorSpace );
color.setHex( hex, SRGBColorSpace );
```

... could become configurable ...

```js
color.setRGB( r, g, b, ColorManagement.workingColorSpace );
color.setHex( hex, ColorManagement.inputColorSpace );
```

Comment (donmccurdy) [reply examples/jsm/loaders/3MFLoader.js:362]: In other words, the loader's reasons for using this value are distinct from our reasons for using it as the default in the setter. @gkjohnson also seems to prefer something explicit. Perhaps this is a good compromise in a loader?

```js
const COLOR_SPACE_3MF = THREE.SRGBColorSpace;

...

colorObject.setStyle( color.substring( 0, 7 ), COLOR_SPACE_3MF );
```

**Comment by donmccurdy** [comment]
@gkjohnson @Mugen87 I've updated the PR, hopefully a good balance of the concerns here?

**Review by Mugen87 — APPROVED** [review]

File: examples/jsm/loaders/LDrawLoader.js [hunk 2]
```diff
@@ -2351,45 +2354,44 @@ class LDrawLoader extends Loader {
 
 		}
 
+		material.color.setStyle( color, COLOR_SPACE_LDRAW );
```
Comment (WestLangley) [inline examples/jsm/loaders/LDrawLoader.js:2357]: Should `color` be named 'hexColor' for clarity?

File: examples/jsm/loaders/LDrawLoader.js [hunk 3]
```diff
@@ -2351,45 +2354,44 @@ class LDrawLoader extends Loader {
...
 		material.depthWrite = ! isTransparent;
-		material.color.convertSRGBToLinear();
 
 		material.polygonOffset = true;
 		material.polygonOffsetFactor = 1;
 
 		if ( luminance !== 0 ) {
 
-			material.emissive.set( material.color ).multiplyScalar( luminance );
+			material.emissive.setStyle( material.color, COLOR_SPACE_LDRAW ).multiplyScalar( luminance );
```
Comment (WestLangley) [inline examples/jsm/loaders/LDrawLoader.js:2368]: Here, `material.color` is a `THREE.Color`. Is this correct?

Comment (donmccurdy) [reply examples/jsm/loaders/LDrawLoader.js]: No that's a mistake, thanks!

File: examples/jsm/loaders/LDrawLoader.js (hunk 2 above)
Comment (donmccurdy) [reply examples/jsm/loaders/LDrawLoader.js]: I'm not sure on this, have clarified fillColor / edgeColor for now.

Comment (WestLangley) [reply examples/jsm/loaders/LDrawLoader.js]: Just not `color`.  :-)

File: examples/jsm/loaders/LDrawLoader.js [hunk 4]
```diff
@@ -2165,8 +2165,8 @@ class LDrawLoader extends Loader {
 		let code = null;
 
 		// Triangle and line colors
-		let fillColor = 0xFF00FF;
-		let edgeColor = 0xFF00FF;
+		let fillColor = '#FF00FF';
+		let edgeColor = '#FF00FF';
```
Comment (WestLangley) [inline examples/jsm/loaders/LDrawLoader.js:2169]: So why was this change required? Will hex not work correctly.

Comment (donmccurdy) [reply examples/jsm/loaders/LDrawLoader.js:2169]: The color value is either the default given here, or a hex string parsed from the file. Rather than trying to choose setHex or setStyle conditionally, I'm ensuring the value is always a hex string.
